pkg/identity/basicallocator/basic_allocator.go

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package basicallocator
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/cilium/cilium/pkg/idpool"
+)
+
+type BasicIDAllocator struct {
+	idPool     idpool.IDPool
+	minIDValue idpool.ID
+	maxIDValue idpool.ID
+}
+
+// NewBasicIDAllocator creates a new BasicIDAllocator for ID values between
+// the provided minIDValue and maxIDValue.
+func NewBasicIDAllocator(minIDValue, maxIDValue idpool.ID) *BasicIDAllocator {
+	return &BasicIDAllocator{
+		idPool:     idpool.NewIDPool(minIDValue, maxIDValue),
+		minIDValue: minIDValue,
+		maxIDValue: maxIDValue,
+	}
+}
+
+func (b *BasicIDAllocator) AllocateRandom() (idpool.ID, error) {
```

Comment:
Are all those methods from some interface?

Reply:
No. They are not made to match any interface.